### PR TITLE
updated README file to state we support current and previous releases

### PR DESCRIPTION
**Pull request by guy9**
@tzach please review

### CODE REVIEW

**Comment by tzach** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ the required compilers, libraries and build tools. Using the frozen toolchain
 allows you to avoid changing anything in your build machine to meet Scylla's
 requirements - you just need to meet the frozen toolchain's prerequisites
 (mostly, Docker or Podman being available).
+Notice that ScyllaDB maintains the current and previous releases of Scylla Open Source.
```
Comment (avikivity) [inline README.md:27]: That's very uninformative. We don't maintain all previous releases.

File: README.md [hunk 2]
```diff
@@ -70,7 +71,7 @@ See [test.py manual](docs/testing.md).
 ## Scylla APIs and compatibility
 By default, Scylla is compatible with Apache Cassandra and its APIs - CQL and
 Thrift. There is also support for the API of Amazon DynamoDB™,
-which needs to be enabled and configured in order to be used. For more
+which needs to be enabled and configured to be used. For more
```
Comment (avikivity) [inline README.md:74]: That doesn't seem related.

File: README.md (hunk 1 above)
Comment (nyh) [reply README.md:27]: Also, I couldn't even figure out what you tried to convey here - that ScyllaDB the company maintains the code?  If that's the intention, why did you choose to write it in the end of the seemingly-unrelated "Build Prerequisites" section,  especially when we already have a section right before it about "What is Scylla"?

File: README.md [hunk 3]
```diff
@@ -80,17 +81,17 @@ and the current compatibility of this feature as well as Scylla-specific extensi
 
 Documentation can be found in [./docs](./docs) and on the
 [wiki](https://github.com/scylladb/scylla/wiki). There is currently no clear
-definition of what goes where, so when looking for something be sure to check
+definition of what goes where, so when looking for something, be sure to check
```
Comment (nyh) [inline README.md:84]: If you're doing such syntax fixes, please title the patch "copy-editing" or "minor fixes", etc., not "state we support current nd previous release".

File: README.md [hunk 4]
```diff
@@ -106,7 +107,7 @@ If you are a developer working on Scylla, please read the [developer guidelines]
 * The [users mailing list] and [Slack channel] are for users to discuss configuration, management, and operations of the ScyllaDB open source.
 * The [developers mailing list] is for developers and people interested in following the development of ScyllaDB to discuss technical topics.
 
-[Users mailing list]: https://groups.google.com/forum/#!forum/scylladb-users
+[Users' mailing list]: https://groups.google.com/forum/#!forum/scylladb-users
```
Comment (nyh) [inline README.md:110]: I think (but I'm not sure, I'm not an MD expert) this needs to be the same as the  [users mailing list] link above. You can't change just one.

**Review by nyh — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (raphaelsc) [reply README.md:110]: you're right @nyh, if you look at resulting file with github ui, you'll notice the link is now broken.

**Comment by tzach** [comment]
@guy9 is this PR till relevant? if not please close
